README.adoc

```diff
@@ -36,30 +38,222 @@ include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/
+    }
+    ...
+}
+
+dependencies {
+    ...
+    compile("org.springframework.cloud:spring-cloud-gcp-starter-pubsub:1.0.0.BUILD-SNAPSHOT")
+    compile("org.springframework.cloud:spring-integration-gcp:1.0.0.BUILD-SNAPSHOT")
+    ...
+}
+```
+
+== Create application files
+
+We'll need a class to include our channel adapter and messaging configuration. Create a
+PubsubApplication class with the @SpringBootApplication header, as is typical with a Spring Boot
+application.
+
+`src/main/java/hello/PubsubApplication.java`
+[source, java]
+----
+@SpringBootApplication
+public class PubsubApplication {
+
+  public static void main(String[] args) throws IOException {
+    SpringApplication.run(PubsubApplication.class, args);
+  }
+
+}
+----
+
+Additionally, since we're building a web application, you can also create a WebAppController class
+to separate between the controller and configuration logic, although they can also live in the same
+place.
+
+`src/main/java/hello/WebAppController.java`
+[source, java]
+----
+@RestController
+public class WebAppController {
+}
+----
+
+We're still missing two files: the HTML and the properties one.
+
+`src/main/resources/static/index.html`
+[source, java]
+----
+include::/complete/src/main/resources/static/index.html[]
+----
+
+`src/main/resources/application.properties`
+[source, java]
+----
+include::/complete/src/main/resources/application.properties[]
+----
+
+== Create an inbound channel adapter
 
-Create a new controller for your Spring application:
+An inbound channel adapter listens to messages from a Google Cloud Pub/Sub subscription and sends
+them to a Spring channel in an application.
 
-`src/main/java/hello/GreetingController.java`
+Instantiating an inbound channel adapter requires a subscriber factory and the name of an existing
+subscription. The Spring Cloud GCP Pub/Sub Boot starter provides a pre-configured subscriber factory
+which you can simply inject as a method argument.
+
+`src/main/java/hello/PubsubApplication.java`
 [source,java]
 ----
-include::complete/src/main/java/hello/GreetingController.java[]
+include::/complete/src/main/java/hello/PubsubApplication.java[tag=messageChannelAdapter]
 ----
 
-NOTE: The above example does not specify `GET` vs. `PUT`, `POST`, and so forth, because `@RequestMapping` maps all HTTP operations by default. Use `@RequestMapping(method=GET)` to narrow this mapping.
+After the channel adapter is instantiated, an output channel where the adapter sends the received
+messages to must be configured. This channel can be explicitly declared through a bean.
 
+`src/main/java/hello/PubsubApplication.java`
+[source,java]
+----
+include::/complete/src/main/java/hello/PubsubApplication.java[tag=pubsubInputChannel]
+----
 
-== Make the application executable
+The `"messages"` subscription must be created before the application is run. New subscriptions can be
+created through the https://console.cloud.google.com/[Google Cloud Console] or, programatically,
```

Comment:
As of the current state of the code a topic named "test" needs to be created as well - I think it would be useful to mention it. 

Also, I would consider renaming both to be more specific like "spring-cloud-gcp-topic" and the subscription to "spring-cloud-gcp-subscription".